woocommerce/woocommerce-ios
# Add In-app Feedback Card

Part of #2537. 

This adds the in-app feedback card in the Today tab for Stats V4. The card currently does not do anything. This is mostly just a UI implementation. This should unblock the other In-app Feedback tasks like #2544 and #2543.  

## Design

Large Font | Light  | Dark
--------|-------|-
![Simulator Screen Shot - iPhone 8 (13 6) (Large Font) - 2020-07-28 at 17 37 59](https://user-images.githubusercontent.com/198826/88739968-238b7e80-d0f9-11ea-832d-0d8b5f523b01.png)        |       ![Simulator Screen Shot - iPhone 8 - 2020-07-28 at 17 35 16](https://user-images.githubusercontent.com/198826/88739961-21292480-d0f9-11ea-8a58-678de3b1f078.png) | ![Simulator Screen Shot - iPhone 8 (13 6) (Dark) - 2020-07-28 at 17 36 29](https://user-images.githubusercontent.com/198826/88739964-22f2e800-d0f9-11ea-8f52-36061e477c8e.png)

We'll ask Garance for a design review when she's back. 

## Changes

The card has to be added as a child of `StoreStatsAndTopPerformersPeriodViewController` so I created it as a `UIViewController`, [`InAppFeedbackCardViewController`](https://github.com/woocommerce/woocommerce-ios/blob/issue/2537-add-in-app-feedback-card/WooCommerce/Classes/ViewRelated/InAppFeedback/InAppFeedbackCardViewController.swift).

The card is only shown when:

- The `inAppFeedback` feature flag is turned on
- If the [`StoreStatsAndTopPerformersPeriodViewController .canDisplayInAppFeedback`](https://github.com/woocommerce/woocommerce-ios/blob/acbefae18c95b9a7d70f9439d159ad73837cdced/WooCommerce/Classes/ViewRelated/Dashboard/Stats%20v4/StoreStatsAndTopPerformersPeriodViewController.swift#L82-L86) is `true`. It's `true` for only the Today tab. 

I'll add unit tests in upcoming PRs since I still need to add another condition for when to display the in-app feedback card. 

## Testing

1. Log in to a store with WC > 4.0
2. Navigate to My Store → Today.
3. Confirm the card is there.
4. Confirm that the card is not visible in the other tabs. 

## Submitter Checklist
